vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/ItemClickEvent.java

```diff
@@ -77,6 +81,8 @@
      */
     public ItemClickEvent(Grid<T> source, boolean fromClient,
             @EventData("event.detail.itemKey") String itemKey,
+            @EventData("event.detail.columnKey") String columnKey,
+            @EventData("event.detail.flowKey") String flowColumnKey,
```

Comment:
Technically adding these args is a breaking change, because it will remove the constructor with the earlier set of arguments. Although these event objects should be constructed by the framework, it's still possible for users to call the constructor.

I'd say that we have to add another constructor with the previous set of arguments, but without the `@EventDatas` to keep backwards compatibility, and hope that Flow is smart enough to pick this constructor with the `@EventDatas`.

@Legioth, any advice for this kind of situation: a new feature which requires adding `@EventData`?

Reply:
@pekam @Legioth since since the first parameter is a string, is it acceptable to pass the flow id of the column in it? I would use some magic connector, like `::` to split it into real item key and column identifier? if `::` is not found, it means there is no column id (backwards compatibility) - just hoping there is no way to have item key itself include `::`, is that possible?

Reply:
That's quite a clever idea, and it should work because `itemKey` is an auto-generated positive integer. But that's also really hacky and I'd like to avoid that.

I still think that another constructor would be the best solution, but would like to wait for @Legioth's opinion, since this is the first time for me to see this kind of situation.

Reply:
You can leave the old constructor signature without any `@EventData` annotations and then add a new one with all of them. At least based on reading the code in `com.vaadin.flow.component.ComponentEventBusUtil.findEventConstructor(Class<T>)`, the constructor without annotated parameter should simply be ignored.